### DISCUSSION TITLE
homeassistant / mqtt control

### DISCUSSION THREAD

**Discussion by kevincw01** [tagyoureit/nodejs-poolController#1037] [Nixie] [answered]
Using nixie standalone and a single pentair vsf pump.  I have mqtt and homeassistant enabled and properly connected.  From dash panel I can turn the pump on and off and set the RPMs.   However in homeassistant I only see a heater control and a couple sensors.  I'd like to turn on/off the pump and change the RPM from home assitant (or mqtt if necessary).  Any tips on how to do that?

**Comment by kevincw01** [comment, marked as answer]
sadly not a very active community on this project.  I tried every option I could think of to enable this and came up empty for the RPM change.  I did find a homeassistant wiki page that shows how to use state/circuits/toggleState to toggle the pump power but it's not explicit so you have to live with whatever is next and not specifically off and on and this is a problem for me.  

So it was time to hack up the source.  It works now but I should clean it up a bit before doing a PR for contribution back to the repository.  Essentially what I did was change the mqttinterface.ts to include a config/pump subscription and the related logic.  I was able to copy the logic that is exposed via express put api on port 4200 that is used between dash and the server.  One mqtt publish changes the rpm and the other can turn the pump off and on.  The latter was already implemented in mqttinterface.ts but not documented on the wiki: use state/circuits/setstate with json params id and isOn.  

Hopefully this comment helps until I get around to the PR

**Reply by tagyoureit** [reply]
Is this the Home Assistant integration you are using - https://github.com/tagyoureit/nodejs-poolController/wiki/Home-Assistant-Setup-Instructions.  It's maintained by community member.

**Reply by Crewski** [reply]
It's been a while since I've made any changes to the code, but here is the custom integration I made with the help of @rstrouse a while back.  Install it via HACS.  https://github.com/Crewski/njsPC-HA.  It doesn't use mqtt, but socketio and http requests (same way the dashPanel operates).
